download/download_test.go

```diff
@@ -652,6 +654,53 @@ func TestOneShotNotLongPollingSwitch(t *testing.T) {
 	}
 }
 
+func TestWarnOnNonBundleContentType(t *testing.T) {
+	ctx := context.Background()
+	fixture := newTestFixture(t)
+	fixture.server.bundles["not-a-bundle"] = bundle.Bundle{}
+
+	updates := make(chan *Update)
+
+	config := Config{}
+	if err := config.ValidateAndInjectDefaults(); err != nil {
+		t.Fatal(err)
+	}
+
+	d := New(config, fixture.client, "/bundles/not-a-bundle").WithCallback(func(_ context.Context, u Update) {
+		updates <- &u
+	})
+	logger := test.New()
+	logger.SetLevel(logging.Debug)
+	d.logger = logger
+
+	d.Start(ctx)
+
+	// Give time for some download events to occur
+	time.Sleep(1 * time.Second)
+
+	u1 := <-updates
+
+	expectLogged := "Content-Type response header set to text/html. " +
+		"Expected one of [application/gzip application/octet-stream]. Possibly not a bundle being downloaded."
+	var found bool
+	for _, entry := range logger.Entries() {
+		if entry.Message == expectLogged {
+			found = true
+			break
+		}
+	}
+
+	if !found {
+		t.Errorf("Expected log entry: %s", expectLogged)
+	}
+
+	if u1.Bundle != nil {
+		t.Fatal("expected no bundle")
+	}
+
+	d.Stop(ctx)
```

Comment:
These tests are often prone to race conditions. Can we Stop earlier, just to ensure that it's safe to read from `logger.Entries()`? I'd expect it to be OK to close right after L681 🤔

Reply:
Yep, moving the stop to where you suggested 👍